ocaml/xapi/xapi_vdi.ml

```diff
@@ -1072,8 +1072,13 @@ let _get_nbd_info ~__context ~self ~get_server_certificate =
     if ips = [] then [] else
     let cert = get_server_certificate ~host in
     let port = 10809L in
-    (* Stopgap measure: use hostname instead of reading a subject out of the cert. *)
-    let subject = Db.Host.get_hostname ~__context ~self:host in
+    let subject = match Certificates.hostnames_of_pem_cert cert with
```

Comment:
I reckon we should fall back to the hostname in case the library fails here with an exception.

Reply:
As the [documentation of the library](http://docs.mirage.io/x509/X509/index.html) states:
>Missing is the handling of certificate revocation lists, online certificate status protocol, some X.509v3 extensions (such as policy and name constraints). The only supported key type is RSA.

I guess in these cases it will just throw an exception - so I guess if we have the above fallback, then we don't have to worry about this too much?

Reply:
Also, in case one host has an incorrect certificate, `get_nbd_info` will fail completely, and not return anything - not sure what the right behaviour is here - maybe that's fine.